cms/tests/test_page_admin.py

```diff
@@ -1912,6 +1912,54 @@ def test_clear_placeholder_marks_page_as_dirty(self):
             self.assertEqual(page.reload().get_publisher_state("en"), PUBLISHER_STATE_DIRTY)
 
 
+    @override_settings(USE_THOUSAND_SEPARATOR=True, USE_L10N=True)
+    def test_page_tree_render_localized_page_ids(self):
+        admin_user = self.get_superuser()
+        root = create_page("home", "nav_playground.html", "fr",
```

Comment:
Potentially something like

    Page.objects.raw('ALTER TABLE cms_page AUTO_INCREMENT=1001;')

In stead of loop. I have not tried this, though.

Reply:
I will test this. My first impression is that there isn't a db-agnostic SQL syntax to fix the auto-icrement, but I might be wrong.

Reply:
https://github.com/django-cms/django-cms/pull/7188/commits/666a88a8449741388668da99287784cf05ced6b2 looks like a practical solution to me. 👍. 

I am also looking at https://djlint.com/docs as a template linter. It would allow to add a rule like

```
- rule:
    name: D034
    message: .pk or .id used without unlocalize
    flags: re.DOTALL|re.I
    patterns:
      - {{\s*[a-zA-Z0-9_.]*.pk\s*}}
      - {{\s*[a-zA-Z0-9_.]*.id\s*}}
```

to find all unlocalized references to database ids. In total I would feel better to add the linter to all our projects to avoid this issue wherever it happens. Then IMHO we could skip the test altogether. Will try with djangocms-frontend. What do you think @marksweb?

Reply:
Here's my try for djangocms-frontend (https://github.com/django-cms/djangocms-frontend/pull/58). Immediatly found ~5 bugs in the templates!!

Reply:
I have now run a linter through the cms code base. There are more breaking templates if page numbers exceed 1000. 
The following files were detected:

```
cms/templates/admin/cms/page/tree/actions_dropdown.html
cms/templates/admin/cms/page/tree/base.html
cms/templates/admin/cms/page/tree/menu.html (already part of this PR)
cms/templates/cms/toolbar/dragitem.html
cms/test_utils/project/templates/integration/plugins/no_custom_model.html 
cms/test_utils/project/templates/pluginapp/plugins/no_custom_model.html
cms/test_utils/project/templates/plugins/no_custom_model.html
```

Can we extend the PR to cover at least the non-testing templates (first 4)? The solution would be to `{% load l10n %}` and append `|unlocalize` to the use of `.pk`.

Reply:
Here's the full lint output:

```
Linting 254/254 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00    


cms/templates/admin/cms/page/tree/actions_dropdown.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 8:25 .pk or .id should only be used with "|unlocalize" {{ page.pk }}
D034 9:30 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 21:25 .pk or .id should only be used with "|unlocalize" {{ page.pk }}
D034 22:30 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 32:33 .pk or .id should only be used with "|unlocalize" {{ page.pk }}
D034 33:30 .pk or .id should only be used with "|unlocalize" {{ node.pk }}

cms/templates/admin/cms/page/tree/base.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 111:50 .pk or .id should only be used with "|unlocalize" {{ site.pk }}
D034 111:111 .pk or .id should only be used with "|unlocalize" {{ site.pk }}
D034 130:51 .pk or .id should only be used with "|unlocalize" {{ site.pk }}
D034 198:36 .pk or .id should only be used with "|unlocalize" {{ tree.site.pk }}
D034 262:32 .pk or .id should only be used with "|unlocalize" {{ tree.site.pk }}

cms/templates/admin/cms/page/tree/menu.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 19:71 .pk or .id should only be used with "|unlocalize" {{ page.id }}
D034 20:13 .pk or .id should only be used with "|unlocalize" {{ page.pk }}
D034 21:18 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 150:84 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 164:37 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 164:61 .pk or .id should only be used with "|unlocalize" {{ page.pk }}
D034 186:58 .pk or .id should only be used with "|unlocalize" {{ node.pk }}
D034 186:82 .pk or .id should only be used with "|unlocalize" {{ page.pk }}

cms/templates/cms/toolbar/dragitem.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 50:76 .pk or .id should only be used with "|unlocalize" {{ plugin.pk }}

cms/test_utils/project/templates/integration/plugins/no_custom_model.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 1:3 .pk or .id should only be used with "|unlocalize" {{ instance.pk }}

cms/test_utils/project/templates/pluginapp/plugins/no_custom_model.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 1:3 .pk or .id should only be used with "|unlocalize" {{ instance.pk }}

cms/test_utils/project/templates/plugins/no_custom_model.html
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D034 1:3 .pk or .id should only be used with "|unlocalize" {{ instance.pk }}

Linted 254 files, found 23 errors.

```

Reply:
@fsbraun done (in f99ad42423db24482d74f2f39412ee455fa51cb9). I didn't find the last three templates you mentioned in `develop`, though!

Reply:
@mbi Excellent work! Thank you so much! I believe you've just fixed an extremely difficult to find bug. 🙌